dev-support/checks/sonar.sh

```diff
@@ -20,4 +20,4 @@ if [ ! "$SONAR_TOKEN" ]; then
   echo "SONAR_TOKEN environment variable should be set"
   exit 1
 fi
-mvn -B verify -DskipShade -DskipTests org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=apache -Dsonar.projectKey=apache_incubator-ratis
+mvn -B verify -DskipShade -DskipTests org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=apache -Dsonar.projectKey=apache-ratis
```

Comment:
I don't think this should be changed, as the project ID at Sonarcloud is still `apache_incubator-ratis`.

https://sonarcloud.io/project/issues?id=apache-ratis -> "The requested project does not exist."
https://sonarcloud.io/project/issues?id=apache_incubator-ratis -> OK

Reply:
Thanks for reviewing, @adoroszlai .

I'm not familiar with the Sonar setup.  Does the project key need to be set up ahead of time somehow, or would it get activated on the first Sonar run under the new key?

Ideally, I'd like to transition to "apache-ratis" and completely remove "incubator" for consistency.

Reply:
We probably need to file an INFRA JIRA to change the Sonarcloud project key.  The earlier INFRA JIRAs are
- https://issues.apache.org/jira/browse/INFRA-19555
- https://issues.apache.org/jira/browse/INFRA-20295